sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/base/BaseViewModel.kt

```diff
@@ -46,8 +43,12 @@ abstract class BaseViewModel(private val dispatchersProvider: DispatchersProvide
         }
     }
 
-    fun execute(coroutineDispatcher: CoroutineDispatcher = dispatchersProvider.io, job: suspend () -> Unit) =
-        viewModelScope.launch(coroutineDispatcher) {
+    protected fun launch(context: CoroutineContext = EmptyCoroutineContext, job: suspend () -> Unit) =
+        viewModelScope.launch(context) {
             job.invoke()
         }
```

Comment:
Here is a little bit of imporvement but I feel like we are over-engineering this part. It is almost the same as calling `viewModelScope.launch` by ourselves. What do you guys think? @nimblehq/android-chapter 
```suggestion
    protected fun launch(context: CoroutineContext = EmptyCoroutineContext, job: suspend CoroutineScope.() -> Unit) =
        viewModelScope.launch(context, block = job)
```

Reply:
@kaungkhantsoe Actually, I tried to follow the current approach to shorten the default execution, then we can get rid of some max line length Detekt issues. I follow your options here @nimblehq/android-chapter  😄